internal/controllers/extension_controller_test.go

```diff
@@ -92,6 +98,152 @@ func TestExtensionReconcile(t *testing.T) {
 	}
 }
 
+func TestExtensionResolve(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.EnableExtensionAPI, true)()
+	ctx := context.Background()
+
+	testCases := []struct {
+		name                    string
+		packageName             string
+		packageVersion          string
+		upgradeConstraintPolicy ocv1alpha1.UpgradeConstraintPolicy
+		wantErr                 error
+		existingApp             *carvelv1alpha1.App
+		assert                  func(*testing.T, ctrl.Result, client.Client, types.NamespacedName)
```

Comment:
It seems the only difference in the `assert` function implementation for each case is where it expects a True or False condition, and the Message. Would it be better to update the test case struct to have the expected condition and message and move the assertions to loop? That way if something changes we don't need to do it for each test case.

Reply:
Updated